pkgs/test_core/lib/src/runner/engine.dart

```diff
@@ -293,6 +302,30 @@ class Engine {
     return success;
   }
 
+  Future<Null> _gatherCoverage(LiveSuiteController controller) async {
+    if (_coverage == null) return;
+
+    final RunnerSuite suite = controller.liveSuite.suite;
+
+    if (!suite.platform.runtime.isDartVM) return;
+
+    // for some reason the observatoryUrl URI doesn't have the query parameters populated
+    final String isolateId =
+        Uri.parse(suite.environment.observatoryUrl.fragment)
+            .queryParameters['isolateId'];
+
+    final Map<String, dynamic> cov = await collect(
+        suite.environment.observatoryUrl, false, false, false, Set(),
+        isolateIds: {isolateId});
+
+    final outfile = File('$_coverage/${suite.path}.vm.json')
+      ..createSync(recursive: true);
```

Comment:
[nit] The coverage directory should already be created at this point (https://github.com/dart-lang/test/pull/1088/files#diff-01e4e27436330f7aaa3e4b86002db314R112) and writing to a file will create it if it doesn't already exist, so you should be able to remove this `createSync` call.

Reply:
The lowest level directory exists, but if I'm outputting a deep file such as `coverage/test/vm/test.dart.vm.json` those other directories don't exist yet (potentially)

Reply:
Ah good point, sorry I missed that.